### ISSUE TITLE
Optional type return, not behave consistency.

### ISSUE THREAD

**Issue by oshriPP** [microsoft/pyright#1255]
**Describe the bug**
Optional type return, not behave  consistency.

**To Reproduce**
```python 
from typing import Optional, Dict

def f() -> Optional[Dict]:
    return {'a': 1}


foo = {}
bar = f()
bar.get('bar')
foo.update(bar)

if bar is None:
    bar = {}
foo.update(bar)

```

```bash
➜  pubplus-bid-automation (master) ✗ pyright tmp.py
Loading configuration file at ./pyrightconfig.json
Searching for source files
Found 1 source file
./tmp.py
  12:1 - error: No overloads for "foo.update(bar)" match parameters
    Argument types: (Dict[Unknown, Unknown] | None) (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
Completed in 0.615sec
```

**Expected behavior**
I expected to get an error also for `bar.get('bar')`,  because `bar` can be `None` and I want to avoid `AttributeError:`
`AttributeError: 'NoneType' object has no attribute 'get'`

**VS Code extension or command-line**
I'm using VS Code extension and cli tool. 
version: 1.1.93

**Comment by erictraut** [comment]
If you enable the "reportOptionalMemberAccess" diagnostic rule, pyright will report this error. It is disabled by default. For details about how to configure this, refer to [this documentation](https://github.com/microsoft/pyright/blob/master/docs/configuration.md).
